packages/material-ui/src/Slider/SliderStyled.js

```diff
@@ -0,0 +1,303 @@
+    }),
+    '&.MuiSlider--active': {
+      boxShadow: `0px 0px 0px 14px ${fade(props.theme.palette.secondary.main, 0.16)}`,
+    },
+  }),
+}));
+
+export const SliderValueLabel = styled(ValueLabel)({
+  // IE 11 centering bug, to remove from the customization demos once no longer supported
+  left: 'calc(-50% - 4px)',
+});
+
+export const SliderMark = styled('span', { shouldForwardProp })((props) => ({
+  position: 'absolute',
+  width: 2,
+  height: 2,
+  borderRadius: 1,
+  backgroundColor: 'currentColor',
+  ...(props.markActive && {
+    backgroundColor: props.theme.palette.background.paper,
+    opacity: 0.8,
+  }),
+}));
+
+export const SliderMarkLabel = styled('span', { shouldForwardProp })((props) => ({
+  ...props.theme.typography.body2,
+  color: props.theme.palette.text.secondary,
+  position: 'absolute',
+  top: 26,
+  transform: 'translateX(-50%)',
+  whiteSpace: 'nowrap',
+  ...(props.orientation === 'vertical' && {
+    top: 'auto',
+    left: 26,
+    transform: 'translateY(50%)',
+  }),
+  '@media (pointer: coarse)': {
+    top: 40,
+    ...(props.orientation === 'vertical' && {
+      left: 31,
+    }),
+  },
+  ...(props.markLabelActive && {
+    color: props.theme.palette.text.primary,
+  }),
+}));
+
+const useThemeProps = (inputProps, ref, name) => {
+  const props = Object.assign({}, inputProps);
+  const { innerRef } = props;
+
+  const contextTheme = useTheme() || defaultTheme;
+
+  const more = getThemeProps({ theme: contextTheme, name, props });
+
+  const theme = more.theme || contextTheme;
+  const isRtl = theme.direction === 'rtl';
+
+  return {
+    ref: innerRef || ref,
+    theme,
+    isRtl,
+    ...more,
+  };
+};
+
+const useThemeClasses = (name) => {
+  const theme = useTheme() || defaultTheme;
+
+  let styleOverrides = {};
+  let variants = [];
+
+  if (theme && theme.components && theme.components[name] && theme.components[name].styleOverrides) {
+    styleOverrides = theme.components[name].styleOverrides;
+  }
+
+  if (theme && theme.components && theme.components[name] && theme.components[name].variants) {
+    variants = theme.components[name].variants;
+  }
+
+  const classes = {};
+
+  Object.keys(styleOverrides).forEach(key => {
+    classes[key] = css(styleOverrides[key]);
+  });
+
+  variants.forEach((definition) => {
+    const key = propsToClassKey(definition.props);
+    if(classes[key]) {
+      classes[key] = cx(classes[key], css(definition.style))
+    } else {
+      classes[key] = css(definition.style);
+    }
+  });
+
+  return classes;
+};
+
+const getComponentProps = (components, componentsProps, name) => {
+  return {
+    as: components[name],
+    ...(componentsProps[name] || {}),
+  };
+};
+
+const Slider = React.forwardRef(function Slider(inputProps, inputRef) {
+  const props = useThemeProps(inputProps, inputRef, 'MuiSlider');
+  const classes = useThemeClasses('MuiSlider');
+  const themeVariantsClasses = useThemeVariants({ ...props, classes }, 'MuiSlider');
```

Comment:
Slider is not really supporting variants, but it is added here for POC that it will work